### PR TITLE
Adding Ingress Controller Labels for Metrics to Ingress Controller Pods

### PR DESCRIPTION
**Pull request by jaiveerk**
# Description

Adding the following labels to pods deployed as part of WAR's ingress controller deployments to be able to query metrics from them without scraping the pods directly. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a breaking change which will impact consuming tool(s)?

Unit tests have been added to ensure that changes to fixtures are limited to what they're intended to be. These labels have also been manually tested in the AKS operator to ensure they work as intended for metrics collection purposes.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by bfoley13 — APPROVED** [review]
Looks clean

**Review by OliverMKing — CHANGES_REQUESTED** [review]

File: pkg/manifests/nginx.go [hunk 1]
```diff
@@ -255,14 +255,19 @@ func newNginxIngressControllerService(conf *config.Config, ingressConfig *NginxI
...
-
-	for k, v := range topLevelLabels {
-		ingressControllerLabels[k] = v
+func addControllerDeploymentLabels(originalLabels map[string]string) map[string]string {
+	tr := make(map[string]string)
+	for k, v := range originalLabels {
+		tr[k] = v
 	}
+	tr["app.kubernetes.io/component"] = "ingress-controller"
+	tr["app.kubernetes.io/managed-by"] = "aks-app-routing-operator"
```
[the remark this answers is not in the document]
Comment (jaiveerk) [reply pkg/manifests/nginx.go:264]: fair point, but check out line 295 - this method also needs to work for pod labels. 

I can just duplicate this code in the newNginxIngressControllerDeployment method to add the component label to the top level labels, or, I suppose if we want to propagate the component label to other metrics, I can make a separate method for that and just call that, I'm just not sure about introducing the component label right now - it's out of scope for this PR and makes its purpose unclear (adding necessary labels for metrics in AKS-operator vs introducing a new labeling scheme that unnecessarily adds complexity).

If you feel strongly about using the component label everywhere I can do that, just given my history of PRs in this repo I'm weary of adding complexity where it isn't immediately necessary.

[the remark this answers is not in the document]
Comment (jaiveerk) [reply pkg/manifests/nginx.go:264]: After offline discussion I'll just add in the component label

**Review by OliverMKing — DISMISSED** [review]

File: pkg/manifests/nginx.go [hunk 2]
```diff
@@ -255,14 +255,24 @@ func newNginxIngressControllerService(conf *config.Config, ingressConfig *NginxI
 	}
 }
 
+func addComponentLabel(originalLabels map[string]string, componentName string) map[string]string {
```
Comment (OliverMKing) [inline pkg/manifests/nginx.go:258]: Since this was generalized it probably makes more sense in `common.go`

**Review by OliverMKing — APPROVED** [review]

**Review by davidgamero — APPROVED** [review]

**Review by bfoley13 — APPROVED** [review]

File: pkg/manifests/nginx_test.go [hunk 3]
```diff
@@ -159,3 +159,16 @@ func TestIngressClassResources(t *testing.T) {
...
+
+func TestMapAdditions(t *testing.T) {
+	testMap := map[string]string{"testkey1": "testval1"}
+	withAdditions := addComponentLabel(testMap, "ingress-controller")
+
+	if withAdditions["testkey1"] != "testval1" {
+		t.Errorf("new map doesn't include original values")
+	}
+
+	if withAdditions["app.kubernetes.io/component"] != "ingress-controller" {
```
[the remark this answers is not in the document]
Comment (jaiveerk) [reply pkg/manifests/nginx_test.go:171]: good call!

**Review by davidgamero — DISMISSED** [review]

**Review by bfoley13 — APPROVED** [review]

File: pkg/manifests/fixtures/nginx/full.json [hunk 4]
```diff
@@ -157,7 +159,8 @@
         "name": "nginx",
         "creationTimestamp": null,
         "labels": {
-          "app.kubernetes.io/managed-by": "aks-app-routing-operator"
+          "app.kubernetes.io/managed-by": "aks-app-routing-operator",
+          "app.kubernetes.io/component": "ingress-controller-cluster-role-binding"
```
[the remark this answers is not in the document]
Comment (jaiveerk) [reply pkg/manifests/fixtures/nginx/full.json]: done, generalized component to ingress-controller

**Review by OliverMKing — APPROVED** [review]

**Review by davidgamero — APPROVED** [review]

**Pull request merged**